### PR TITLE
Create row and slot functions

### PR DESCRIPTION
**Pull request by YochayCO**

### CODE REVIEW

File: board.pl [hunk 1]
```diff
@@ -53,37 +55,78 @@
...
+letter_to_row(L, I):-
+    char_code(L, LCode),
+    I is LCode - 64.
 
-% TODO
-% % matches slot's color, depends on board state
-% % slot_color(Board, Slot, Color)
-% slot_color(Board, slot(Num, Letter), color(Color)).
+% matches slot content, depends on board state
+slot_by_index(BoardState, RowIndex, ColIndex, Slot):-
```
Comment (adiva6) [inline board.pl:65]:
```suggestion
slot_by_index(BoardState, RowIndex, ColIndex, SlotType):-
```

File: board.pl [hunk 2]
```diff
@@ -53,37 +55,78 @@
...
 
+% matches border slots, depends on board state (though this remains static throughout the game)
+border(BoardState, RowIndex, ColIndex):-
+    slot_by_index(BoardState, RowIndex, ColIndex, -1).
 
-% TODO
-% % matches all slots of specified row
-% row(slot(Num, Letter), axis(X, Y), Slots).
+% matches all slots of specified horizontal row
+horizontal_row(BoardState, RowIndex, Slots):-
```
Comment (adiva6) [inline board.pl]: I see that here you include the '-1' slots while in other predicates you don't (for example in diagonal_row you skip those slots). Maybe you should make it the same in all predicates

File: board.pl [hunk 3]
```diff
@@ -1,3 +1,4 @@
+:- dynamic board_size/1.
```
Comment (adiva6) [inline board.pl:1]: Suggestion for better naming:
horizontal_row -> row
vertical_row -> column
diagonal_row -> diagonal
What do you think?

**Review by adiva6 — CHANGES_REQUESTED** [review]

File: board.pl (hunk 2 above)
Comment (YochayCO) [reply board.pl]: I do include them also in the diagonal row.
Not sure if I changed it today,
Anyways, we need at least one '-1' in every row we fetch from the board, for the checks

File: board.pl (hunk 1 above)
Comment (YochayCO) [reply board.pl:65]: Whoa yeah that's better, gonna have to change just about everything in the file XD

File: board.pl (hunk 3 above)
Comment (YochayCO) [reply board.pl:1]: sure
and we can name the x,y,z like this as well if you want

File: board.pl [hunk 4]
```diff
@@ -53,59 +57,137 @@
         axis(A, direction(W,V))
     ).
 
-% TODO
-% slot(Num, Letter)
+letter_to_row(L, I):-
```
Comment (adiva6) [inline board.pl]: letter_to_col?

File: board.pl (hunk 3 above)
Comment (adiva6) [inline board.pl:1]: Also, Try to avoid naming variables with just one letter. For example:
letter_to_row(L, I) should be renamed to Letter, Index.

**Review by adiva6 — APPROVED** [review]

**Pull request merged**